analysisservice/src/main/java/nu/yona/server/analysis/service/ActivityCacheService.java

```diff
@@ -26,17 +26,16 @@
 {
 	@Cacheable(key = "{#userAnonymizedId,#deviceAnonymizedId,#goalId}")
 	@Transactional
-	public ActivityDto fetchLastActivityForUser(UUID userAnonymizedId, UUID deviceAnonymizedId, UUID goalId)
+	public Optional<ActivityDto> fetchLastActivityForUser(UUID userAnonymizedId, UUID deviceAnonymizedId, UUID goalId)
 	{
 		List<DayActivity> lastDayActivityList = DayActivity.getRepository()
 				.findLast(userAnonymizedId, goalId, PageRequest.of(0, 1)).getContent();
 		if (lastDayActivityList.isEmpty())
 		{
-			return null;
+			return Optional.empty();
 		}
-		Activity lastActivity = lastDayActivityList.get(0).getLastActivity(deviceAnonymizedId);
-		// this can be the case when a DayActivity is created with inactivity
-		return lastActivity == null ? null : ActivityDto.createInstance(lastActivity);
+		// Return activity DTO, or empty, in case the DayActivity is created with inactivity
+		return lastDayActivityList.get(0).getLastActivity(deviceAnonymizedId).map(entity -> ActivityDto.createInstance(entity));
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Replace this lambda with a method reference. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](http://10.127.10.107:9000/coding_rules#rule_key=squid%3AS1612)